Rename `CanvasItem.update()` to `queue_redraw()`

Mickeon opened the pull request:
Closes https://github.com/godotengine/godot-proposals/issues/5190.

As brought up in https://github.com/godotengine/godot/issues/54161#issuecomment-962015533, https://github.com/godotengine/godot/issues/54161#issuecomment-1034903362 and [Should CanvasItem.`update()` be renamed to `request_redraw()` or  `redraw()`](https://github.com/godotengine/godot-proposals/discussions/5154).

Affects a lot of classes. This PR is absolutely, utterly **GARGANTUAN**.

I very thoroughly checked signal connections and deferred calls to this method, `add_do_method`/`add_undo_method` calls, and so on. Just a few comments have also been changed to say "redraw". It's ENTIRELY possible I still missed something, but the engine compiles, at least!

Also renames the internal `_update_callback()` to `_redraw_callback()` for consistency.

In **CPUParticles2D**, there was a private variable with the same name. For now, this has been renamed to `do_redraw`. It's either that, or something close. I haven't exactly understood what it's meant for.

~This PR was previously renaming it to redraw()~

Mickeon committed:
Rename `CanvasItem.update()` to `queue_redraw()`

Affects a lot of classes. Very thoroughly checked signal connections and deferred calls to this method, add_do_method/add_undo_method calls, and so on.

Also renames the internal `_update_callback()` to `_redraw_callback()` for consistency.

Just a few comments have also been changed to say "redraw".

In CPUParticles2D, there was a private variable with the same name. It has been renamed to `do_redraw`.

Mickeon commented:
There are occasions where `update()` is actually called externally by a parent. For example:
https://github.com/godotengine/godot/blob/master/editor/plugins/tiles/tile_data_editors.cpp#L701#L714

By the naming conventions I've seen, is it still a good idea to rename the internal method `_update_callback`?

YuriSizov commented:
> By the naming conventions I've seen, is it still a good idea to rename the internal method `_update_callback`?

What do you mean? We're renaming the public method to make it clearer what it does, so the internal one needs to be renamed as well to avoid any confusion for engine developers. The name would be something `_redraw_callback`, so I don't see any problem with that.

Mickeon commented:
Oh I see! I assumed the starting underscore was mostly reserved for virtual or private methods.

YuriSizov commented:
> Oh I see! I assumed the starting underscore was mostly reserved for virtual or private methods.

`_update_callback` is and `_redraw_callback` should still remain a private method.

Mickeon commented:
Oooh I see, I understand it all now. Thank you very much

Mickeon commented:
![image](https://user-images.githubusercontent.com/66727710/186942073-98652732-6a64-4bf8-9d80-c3883161c613.png)

Updated the PR to use `queue_redraw`. To be honest, I do still prefer `redraw` over it, because for being so commonly utilised in the codebase it may be bit annoying to see, and the documentation of this method I updated goes into greater detail on how it works, but I've seen a few methods around with similar names and they *do* actually redraw on the spot, so... I suppose it's not worth splitting hair for.

YuriSizov reviewed (approved):
I am tentatively ready to approve this, but it needs another rebase and quick look in case you accidentally renamed stuff you shouldn't have or missed stuff that you should've renamed (though unlikely if you've only did the bulk of changes in various cpp files).

Edit: Also, you didn't rename `_update_callback` to `_redraw_callback`?

YuriSizov commented:
> Updated the PR to use `queue_redraw`. To be honest, I do still prefer `redraw` over it, because for being so commonly utilised in the codebase it may be bit annoying to see, and the documentation of this method I updated goes into greater detail on how it works, but I've seen a few methods around with similar names and they _do_ actually redraw on the spot, so... I suppose it's not worth splitting hair for.

I think it's important to indicate that the method has no immediate effect. We use `queue_*` for other method names as well, including widely utilized `queue_free()`, so it's not a foreign concept.

Mickeon commented:
> Edit: Also, you didn't rename `_update_callback` to `_redraw_callback`?

I thought of doing it in another PR but it doesn't hurt to just do it now at this point.

Mickeon commented:
All done, I had checked one more time.

YuriSizov reviewed (approved):
Changeset seems good now. Didn't check every single line, but in principle it should be correct if the CI passes

YuriSizov commented:
![image](https://user-images.githubusercontent.com/11782833/187440857-9c253e11-99a6-4def-90c8-18a70c93c72f.png)
